pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -407,7 +407,9 @@ def training_type_plugin(self) -> TrainingTypePlugin:
         # attach checkpoint plugin to the training type plugin
         if self._checkpoint_io is not None:
             self._training_type_plugin.checkpoint_io = self._checkpoint_io
-        self._training_type_plugin._precision_plugin = self.precision_plugin
+        precision_plugin = self.precision_plugin
+        if precision_plugin is not None:
```

Comment:
But it could never be None?

Reply:
Yeah, as we discussed above. But consider all the refactor happening now, feel safer have a none check? 
Training_type_plugin(precision_plugin=None) will have default value, but training_type_plugin.precision_plugin will have not default value.